### PR TITLE
Wasm SDK packed as a nuget package

### PR DESCRIPTION
**Pull request by maraf**
Move bits of Wasm SDK that are expected to change more frequently to the runtime repo.

- Add `Microsoft.NET.Sdk.WebAssembly.Pack` implicit package reference if `UsingMicrosoftNETSdkWebAssembly=true`.
- Provide `_WebAssemblyPropsFile` and `_WebAssemblyTargetsFile` properties to enable overriding props and targets.
- Move tests for Wasm SDK tasks to runtime repo. Leave tests checking boot.json in place at the moment.
- Return some default property values back to Blazor SDK to support .NET 5.

Contributes to https://github.com/dotnet/runtime/issues/81367

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs [hunk 1]
```diff
@@ -688,13 +703,13 @@ private enum ToolPackType
             }
 
             // ILCompiler and ILLink have RID-agnostic build packages that contain MSBuild targets.
-            if (toolPackType is ToolPackType.ILCompiler or ToolPackType.ILLink)
+            if (toolPackType is ToolPackType.ILCompiler or ToolPackType.ILLink or ToolPackType.WebAssemblySdk)
```
Comment (lewing) [inline src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs:706]: Not a strong preference here but it seems like
```suggestion
            if (toolPackType is not ToolPackType.Crossgen2)
```
is slightly cleaner

**Review by lewing — COMMENTED** [review]
This is looking good, what else is left?

**Comment by maraf** [comment]
> This is looking good, what else is left?

**Comment by lewing** [comment]
@rainersigwald @tmat @mthalman I'm having a little trouble following all the flow here but are we expecting the failures here to be resolved once everthing flows? https://github.com/dotnet/sourcelink/issues/1007

**Comment by tmat** [comment]
@lewing I think so.

**Comment by lewing** [comment]
triggered https://github.com/dotnet/sdk/pull/31953 but also pulled the bump in here to verify

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
@javiercn @MackinnonBuck any objections?

**Review by javiercn — APPROVED** [review]

**Pull request merged**